lock.go

```diff
@@ -0,0 +1,244 @@
+)
+
+type Lock interface {
+	Lock() bool
+	UnLock(force bool) bool
+}
+
+type NVLocker struct {
+	name          string
+	objMgr        *ObjectManager
+	LockEA        string
+	LockTimeoutEA string
+}
+
+func (l *NVLocker) createLockRequest() *MultiRequest {
+
+	req := NewMultiRequest(
+		[]*RequestBody{
+			&RequestBody{
+				Method: "GET",
+				Object: "networkview",
+				Data: map[string]interface{}{
+					"name":         l.name,
+					"*" + l.LockEA: freeLockVal,
+				},
+				Args: map[string]string{
+					"_return_fields": "extattrs",
+				},
+				AssignState: map[string]string{
+					"NET_VIEW_REF": "_ref",
+				},
+				Discard: true,
+			},
+			&RequestBody{
+				Method: "PUT",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Data: map[string]interface{}{
+					"extattrs+": map[string]interface{}{
+						l.LockEA: map[string]string{
+							"value": l.objMgr.TenantID,
+						},
+						l.LockTimeoutEA: map[string]int32{
+							"value": int32(time.Now().Unix()),
+						},
+					},
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "GET",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Args: map[string]string{
+					"_return_fields": "extattrs",
+				},
+				AssignState: map[string]string{
+					"DOCKER-ID": "*" + l.LockEA,
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "STATE:DISPLAY",
+			},
+		},
+	)
+
+	return req
+}
+
+func (l *NVLocker) createUnlockRequest(force bool) *MultiRequest {
+
+	getData := map[string]interface{}{"name": l.name}
+	if !force {
+		getData["*"+l.LockEA] = l.objMgr.TenantID
+	}
+
+	req := NewMultiRequest(
+		[]*RequestBody{
+			&RequestBody{
+				Method: "GET",
+				Object: "networkview",
+				Data:   getData,
+				Args: map[string]string{
+					"_return_fields": "extattrs",
+				},
+				AssignState: map[string]string{
+					"NET_VIEW_REF": "_ref",
+				},
+				Discard: true,
+			},
+			&RequestBody{
+				Method: "PUT",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Data: map[string]interface{}{
+					"extattrs+": map[string]interface{}{
+						l.LockEA: map[string]string{
+							"value": freeLockVal,
+						},
+					},
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "PUT",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Data: map[string]interface{}{
+					"extattrs-": map[string]interface{}{
+						l.LockTimeoutEA: map[string]interface{}{},
+					},
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "GET",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Args: map[string]string{
+					"_return_fields": "extattrs",
+				},
+				AssignState: map[string]string{
+					"DOCKER-ID": "*" + l.LockEA,
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "STATE:DISPLAY",
+			},
+		},
+	)
+
+	return req
+}
+
+func (l *NVLocker) Lock() bool {
```

Comment:
OK... I ignored the fact that there is a Locker interface... My bad...

In which case, I think the NetworkViewLocker implementation is problematic. The whole point of a interface is so that any implementation of that interface has similar behavior.

Specifically, a piece of code that utilitize the Locker interface would call Lock() method to create the lock, and, likewise, Unlock() to release the lock - irrespective of the implementation. The way the NetworkViewLocker  is implemented, this wouldn't work.

So, I think you need to refactor the NetworkViewLocker so that it meets that expectation.